server/services/recommendation_service.py

```
@@ -49,9 +47,10 @@ async def recommend_players(self, player_ids: List[int]) -> List[PlayerSearchRes
         
         Placeholder orchestration method for future recommendation flow."""
 
-        start_time = time.perf_counter()
+        if len(player_ids) < 9:
+            raise InputValidationError("A valid roster must contain at least 9 players.")
 
-        logger.info("Starting recommendations for roster with %d players", len(player_ids))
+        start_time = time.perf_counter()
```

Ruff (0.14.5)

Local variable `start_time` is assigned to but never used

Remove assignment to unused variable `start_time`

(F841)